### PR TITLE
:wrench: chore(slo): Add slos for "View Submission" for Slack

### PR DESCRIPTION
**Pull request by iamrajjoshi**
"View Submission" Represents the user finishing the "Resolve" or "Archive" modal. we currently don't have slos for this and its affecting my ability to debug issues with the modal.

### CODE REVIEW

File: src/sentry/integrations/messaging/metrics.py [hunk 1]
```diff
@@ -84,15 +85,15 @@ class MessageCommandHaltReason(StrEnum):
     TEAM_NOT_LINKED = "team_not_linked"
     INSUFFICIENT_ROLE = "insufficient_role"
 
-    def __str__(self) -> str:
```
Comment (iamrajjoshi) [inline src/sentry/integrations/messaging/metrics.py:87]: StrEnum so don't need

File: src/sentry/integrations/slack/webhooks/action.py [hunk 2]
```diff
@@ -364,105 +376,117 @@ def _handle_group_actions(
...
+                try:
+                    webhook_client = WebhookClient(private_metadata["orig_response_url"])
+                    webhook_client.send(
+                        blocks=blocks.get("blocks"), delete_original=False, replace_original=True
+                    )
+                    metrics.incr(
+                        SLACK_WEBHOOK_GROUP_ACTIONS_SUCCESS_DATADOG_METRIC,
+                        sample_rate=1.0,
+                        tags={"type": "submit_modal"},
+                    )
```
Comment (cathteng) [inline src/sentry/integrations/slack/webhooks/action.py:440]: delete this?

**Review by cathteng — APPROVED** [review]

File: src/sentry/integrations/slack/webhooks/action.py (hunk 2 above)
Comment (iamrajjoshi) [reply src/sentry/integrations/slack/webhooks/action.py:440]: ope ur right, need to add todo

File: tests/sentry/integrations/slack/webhooks/actions/test_status.py [hunk 3]
```diff
@@ -255,10 +255,12 @@ def test_archive_issue_until_escalating(self, mock_tags, mock_record):
...
-        assert len(mock_record.mock_calls) == 2
-        start, halt = mock_record.mock_calls
-        assert start.args[0] == EventLifecycleOutcome.STARTED
-        assert halt.args[0] == EventLifecycleOutcome.SUCCESS
+        assert len(mock_record.mock_calls) == 4
+        start_1, success_1, start_2, success_2 = mock_record.mock_calls
+        assert start_1.args[0] == EventLifecycleOutcome.STARTED
+        assert success_1.args[0] == EventLifecycleOutcome.SUCCESS
+        assert start_2.args[0] == EventLifecycleOutcome.STARTED
+        assert success_2.args[0] == EventLifecycleOutcome.SUCCESS
```
Comment (GabeVillalobos) [inline tests/sentry/integrations/slack/webhooks/actions/test_status.py:263]: Do we have tests for the other dialog types? If so, could we also add metric assertions to those as well please?

Comment (iamrajjoshi) [reply tests/sentry/integrations/slack/webhooks/actions/test_status.py:263]: yep i am waiting for tests to fail, thinking there are other places that are missing

**Pull request merged**